Sources/Basic/FileSystem.swift

```diff
@@ -118,6 +118,11 @@ public protocol FileSystem {
     //
     // FIXME: This is obviously not a very efficient or flexible API.
     mutating func writeFileContents(_ path: AbsolutePath, bytes: ByteString) throws
+
+    /// Open an output stream to the given file path.
+    ///
+    /// - Returns: stream to the given file path.
+    func openFileStream(_ path: AbsolutePath) throws -> FileOutputByteStream
```

Comment:
Also, I would probably split up the changes to add the new file-based implementations to OutputByteStream from the changes to add FileSystem support. The file-based OutputByteStream implementations can be used independently of FileSystem (even though most of the higher level SwiftPM code shouldn't).